src/www/ui/api/Helper/DbHelper.php

```diff
@@ -356,6 +356,78 @@ public function getJobs($id = null, $limit = 0, $page = 1, $uploadId = null)
     return [$jobs, $totalResult];
   }
 
+  /**
+   * @brief Get the recent jobs created by an user.
+   *
+   * If a limit is passed, the results are trimmed. If an ID is passed, the
+   * information for the given id is only retrieved.
+   *
+   * @param integer $id       Set to get information of only given job id
+   * @param integer $uid      Set to get information of only given user's ID
+   * @param integer $limit    Set to limit the result length
+   * @param integer $page     Page number required
+   * @param integer $uploadId Upload ID to be filtered
+   * @return array[] List of jobs at first index and total number of pages at
+   *         second.
+   */
+  public function getUserJobs($id = null, $uid=null, $limit = 0, $page = 1, $uploadId = null)
+  {
+    $jobSQL = "SELECT job_pk, job_queued, job_name, job_upload_fk," .
+      " job_user_fk, job_group_fk FROM job WHERE job_user_fk='$uid'";
+    $totalJobSql = "SELECT count(*) AS cnt FROM job WHERE job_user_fk='$uid'";
+
+    $filter = "";
+    $pagination = "";
+
+    $params = [];
+    $statement = __METHOD__ . ".getUserJobs";
+    $countStatement = __METHOD__ . ".getJobCount";
+    if ($id == null) {
+      if ($uploadId !== null) {
+        $params[] = $uploadId;
+        $filter = "WHERE job_upload_fk = $" . count($params);
+        $statement .= ".withUploadFilter";
+        $countStatement .= ".withUploadFilter";
+      }
+    } else {
+      $params[] = $uid;
+      $filter = "WHERE job_user_fk = $" . count($params);
```

Comment:
I am not sure `$uid` should be used here if the `if ($id == null)` checks `$id`.

```suggestion
      $params[] = $id;
      $filter = "WHERE job_pk = $" . count($params);
```